src/accumulator/proof.rs

```diff
@@ -289,10 +301,23 @@ impl Proof {
                 if let Some((next_pos, next_hash)) = row_nodes.peek() {
                     // Is the next node our sibling? If so, we should be hashed together
                     if util::is_right_sibling(pos, *next_pos) {
-                        let hash = types::parent_hash(&hash, &next_hash);
-                        let next_to_prove = util::parent(pos, total_rows);
-
-                        Proof::sorted_push(&mut nodes, (next_to_prove, hash));
+                        // There are three possible cases: the current hash is null,
+                        // and the sibling is present, we push the sibling to targets.
+                        // If The sibling is null, we push the current node.
+                        // If none of them is null, we compute the parent hash of both siblings
+                        // and push this to the next target.
+                        if hash == sha256::Hash::default() {
+                            let next_to_prove = util::parent(pos, total_rows);
```

Comment:
`next_to_prove` is always going to be `util::parent(pos, total_rows)` no matter what the hash is. And it's just a u64 so it's on the stack too.

Just declaring it one right after the `is_root_position` would be better in my opinion.